### ISSUE TITLE
`ERR_REQUIRE_ESM` error code is missing errors resulting from `require`ing esm files

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#16913]
Babel tests this here:

https://github.com/babel/babel/blob/859daa3c185d18901106d8e6c39898ba1ce8e8f4/packages/babel-core/src/config/files/module-types.ts#LL39C25-L39C40

The error type mapping is not working correctly between Rust and JS here:

```
Error 'could not find npm package for 'file:///Users/lucacasonato/projects/local/babeltest/babel.config.js'' contains boxed error of unknown type:
  "could not find npm package for 'file:///Users/lucacasonato/projects/local/babeltest/babel.config.js'"
Error: could not find npm package for 'file:///Users/lucacasonato/projects/local/babeltest/babel.config.js'
    at Object.Module._extensions..js (deno:ext/node/02_require.js:745:28)
    at Module.load (deno:ext/node/02_require.js:643:34)
    at Function.Module._load (deno:ext/node/02_require.js:500:14)
    at Module.require (deno:ext/node/02_require.js:665:21)
    at require (deno:ext/node/02_require.js:799:18)
    at startHiding - secret - don't use this - v1 (file:///Users/lucacasonato/Library/Caches/deno/npm/registry.npmjs.org/@babel/core/7.20.5/lib/errors/rewrite-stack-trace.js:55:12)
    at loadCjsDefault (file:///Users/lucacasonato/Library/Caches/deno/npm/registry.npmjs.org/@babel/core/7.20.5/lib/config/files/module-types.js:77:69)
    at loadCjsOrMjsDefault (file:///Users/lucacasonato/Library/Caches/deno/npm/registry.npmjs.org/@babel/core/7.20.5/lib/config/files/module-types.js:55:16)
    at loadCjsOrMjsDefault.next (<anonymous>)
    at readConfigJS (file:///Users/lucacasonato/Library/Caches/deno/npm/registry.npmjs.org/@babel/core/7.20.5/lib/config/files/configuration.js:86:47)
```

The first message is logged because the error type in Rust has no JS mapping - this is probably also why the code is missing.

**Comment by bartlomieju** [comment]
CC @dsherret I think this is most likely because of use of `bail!`

**Comment by lucacasonato** [comment]
@dsherret any updates on this? just ran into it again?

**Comment by dsherret** [comment]
I didn't properly fix this in #19853. It needs to have the code set.
